src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/FlatFWTest.java

```diff
@@ -133,165 +212,171 @@ public void shouldFailToSetString1BeforeFixed1() throws Exception
+    public void shouldFailToSetString2BeforeString1() throws Exception
     {
         expectedException.expect(IllegalStateException.class);
         expectedException.expectMessage("string1");
-        flatRW.wrap(buffer, 0, 100)
-            .fixed1(10)
-            .build();
+        setAllRequiredBuilderFields(flatRW.wrap(buffer, 0, 100), INDEX_STRING1)
+                .string2("value1");
     }
 
     @Test
     public void shouldFailToBuildWhenString2NotSet() throws Exception
     {
         expectedException.expect(IllegalStateException.class);
         expectedException.expectMessage("string2");
-        flatRW.wrap(buffer, 0, 100)
-            .fixed1(10)
-            .fixed2(111)
-            .string1("value1")
-            .fixed3(33)
-            .build();
+        setAllRequiredBuilderFields(flatRW.wrap(buffer, 0, 100), INDEX_STRING2)
+                .build();
     }
 
-    @Test
-    public void shouldSetAllValues() throws Exception
+    private void checkAllBuilderValuesBySetterType(SetterType setterType)
     {
-        flatRW.wrap(buffer, 0, buffer.capacity())
-                .fixed1(10)
-                .fixed2(20)
-                .string1("value1")
+        final int offset = 11;
+        int expectedLimit = setAllBufferValues(expected, offset);
+
+        Builder builder = flatRW.wrap(buffer, offset, expectedLimit);
+
+        builder.fixed1(10);
+        builder.fixed2(20);
+
+        switch (setterType)
+        {
+            case STRING: builder.string1("value1")
                 .fixed3(30)
-                .string2("value2")
-                .build();
-        flatRO.wrap(buffer,  0,  100);
-        assertEquals(10, flatRO.fixed1());
-        assertEquals(20, flatRO.fixed2());
-        assertEquals("value1", flatRO.string1().asString());
-        assertEquals(30, flatRO.fixed3());
-        assertEquals("value2", flatRO.string2().asString());
+                .string2("value2");
+                break;
+            case STRING_FW:
+                StringFW value1 = stringRW.wrap(valueBuffer,  offset, expectedLimit)
+                        .set("value1", UTF_8)
+                        .build();
+                StringFW value2 = stringRW.wrap(valueBuffer,  offset, expectedLimit)
+                        .set("value2", UTF_8)
+                        .build();
+                builder.string1(value1)
+                    .fixed3(30)
+                    .string2(value2);
+                break;
+            case BUFFER:
+                builder.string1(valueBuffer, 0, 6)
+                        .fixed3(30)
+                        .string2(valueBuffer, 10, 6);
+        }
+
+        int limit = builder.build().limit();
+
+        assertEquals(expectedLimit, limit);
+        assertEquals(expected, buffer);
     }
 
-    @Test
-    public void shouldSetStringValuesUsingStringFW() throws Exception
+    static int setRequiredBuilderValues(FlatFW.Builder builder)
```

Comment:
Either remove or rename it to setAllRequiredFields and have it call the other method (setRequiredFields) passing in Integer.MAX_VALUE